src/components/GraphSidebar/Elements/Patterns/Pattern.svelte

```diff
@@ -0,0 +1,109 @@
+<script lang="ts">
+  import Fa from "svelte-fa/src/fa.svelte";
+  import { faLocation, faEye, faEyeSlash } from "@fortawesome/free-solid-svg-icons";
+  import { createEventDispatcher } from "svelte";
+  import type { PatternType } from "../../types";
+
+  export let pattern: PatternType;
+
+  const dispatch = createEventDispatcher();
+  let currentNodeIdHideSelected: string | undefined = undefined;
+
+  const centerCameraOnNode = (patternId: string) => {
+    dispatch("centerNode", {
+      patternId: patternId,
+    });
+  };
+
+  const hideNodeNeighbors = (e: Event, patternId: string) => {
+    const args = {
+      newPatternId: patternId,
+      currentPatternId: currentNodeIdHideSelected,
+    };
+    currentNodeIdHideSelected = currentNodeIdHideSelected === patternId ? undefined : patternId;
+    dispatch("hideNodeNeighbors", args);
+  };
+</script>
+
+{#if pattern.children?.length > 0}
+  <!-- accordion -->
+  <div class="{`flex ${pattern.classes}`}">
+    <candy-accordion class="w-full" label="{`${pattern.name}`}" minimizeOptions="{true}">
+      <button slot="options" on:click|stopPropagation="{() => centerCameraOnNode(pattern.id)}">
+        <Fa icon="{faLocation}" />
+      </button>
+      <button slot="options" on:click|stopPropagation="{(e) => hideNodeNeighbors(e, pattern.id)}">
+        <Fa icon="{currentNodeIdHideSelected === pattern.id ? faEyeSlash : faEye}" />
+      </button>
+      {#each pattern.children as childPattern}
+        <svelte:self pattern="{childPattern}" on:centerNode on:hideNodeNeighbors />
+      {/each}
+    </candy-accordion>
+  </div>
+{:else}
+  <!-- sidebar-element -->
+  <div class="{`flex ${pattern.classes}`}">
+    <candy-sidebar-element class="w-full" label="{`${pattern.name}`}" minimizeOptions="{true}">
+      <button slot="options" on:click|stopPropagation="{() => centerCameraOnNode(pattern.id)}">
```

Comment:
Maybe I confuse with Vue.js, but I guess you can write : 
```suggestion
      <button slot="options" on:click|stopPropagation="{centerCameraOnNode(pattern.id)}">
```

Reply:
I can't because I am giving a parameter. It would work in this case: `on:click|stopPropagation="{centerCameraOnNode}"`